SU-HSDO/suhumsci
# chore(STN-932): Refine backstop tests

# READY FOR REVIEW

## Summary
This pull request modifies that javascript that generates our backstop.json file to test two mirrored theme sites for Traditional and Colorful. Both sites share the same set of demo pages and content, the only difference being the theme used.

Backstop will compare pages on the staging and dev environment and generate a page comparing images. Any visual differences between pages will be marked as a failed test with differences highlighted in magenta.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. Navigate to the Humsci Basic repository.
2. Run `npm run backstop:init` to generate the backstop.json artifact.
3. Run `npm run backstop:reference` to generate the reference images from staging. This may take several minutes to complete.
4. Run `npm run backstop:test` to generate the test images from dev and then compare with reference. This may take several minutes to complete.
5. Once the test is finished running your browser should open with a test page to review.

If you encounter any errors when running the backstop commands complete the following:
- [ ] Delete the `node_modules` directory.
- [ ] Run `npm cache clean --force` to clear the npm cache.
- [ ] Run `npm install` to reinstall node modules.
- [x] Go through the Steps to Test and tasks should run without errors.
- [x] Confirm that there are no usernames/passwords being committed to the repo.
- [x] Confirm that the generated backstop.json file is not being committed to the repo (this file includes usernames and passwords).
- [x] Confirm that the generate backstop data is not being committed to the repo.
- [x] Confirm that the test page includes both reference and test images, even if some of the images are of a 404 page there should not be any backstop errors.
- [x] Pick a test URL and confirm that there are test images for both Colorful and Traditional as well as at 4 different screen sizes. There should be 8 tests total for each a test URL.

_Note: It's ok that the tests fail at this point. We have not updated the data on the dev environment to test. The important thing is that images are generated and tests run._

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
